staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1445,6 +1445,11 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 				}
 			}
 		}
+
+		// TODO : Remove RegisterStorageCleanup below when PR
+		// https://github.com/kubernetes/kubernetes/pull/50690
+		// merges as that shuts down storage properly
+		RegisterStorageCleanup(e.DestroyFunc)
```

Comment:
if CompleteWithOptions isn't called, or if s.Storage.Storage != nil, this won't register, right?

Reply:
I guess the same was already true of where we were registering inside the decorator

Do our destroy functions tolerate being called twice?

Reply:
> Do our destroy functions tolerate being called twice?

Not everywhere:
- raw etcd3 storage can be called multiple times:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go#L277
- cacher can be stopped multiple times: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1008

- the stop from goroutine from startObservingCount can't be called twice:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1471

Fixed the last one the same way it was previously addressed for etcd3 storage:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go#L277